sale_comment_template/views/base_comment_template_view.xml

```diff
@@ -5,11 +5,8 @@
         <field name="type">ir.actions.act_window</field>
         <field name="res_model">base.comment.template</field>
         <field name="view_mode">tree,form</field>
-        <field name="domain">[('model_ids.model', '=', 'sale.order')]</field>
-        <field
-            name="context"
-            eval="{'default_model_ids': [(4, ref('sale.model_sale_order'))]}"
-        />
+        <field name="domain">[('models', '=', 'sale.order')]</field>
```

Comment:
This would only find templates that are _only_ for `sale.order`s.

According to https://github.com/OCA/purchase-reporting/commit/84a78d6c648fad422746f8eb7e504e2f60f2bd08, this
```suggestion
        <field name="domain">[('model_ids', '=', 'sale.order')]</field>
```
should allow to find `sale.order`'s templates when they are listed along with other models' templates (as it should have been before this change), would you mind including it?
In the commit you also have an example of how to implement a test case, it would be great if you could include that too.